### PR TITLE
Flink: dynamically generate data for upsert UT

### PR DESCRIPTION
**Pull request by yittg**
The second part metioned in https://github.com/apache/iceberg/pull/4532#pullrequestreview-937574093
cc @kbendick

### CODE REVIEW

**Comment by yittg** [comment]
Tests fail due to #4533, wait it being merged.

File: flink/v1.13/flink/src/test/java/org/apache/iceberg/flink/TestFlinkUpsert.java [hunk 1]
```diff
@@ -210,33 +189,7 @@ public void testPrimaryKeyFieldsAtBeginningOfSchema() {
...
-
-      sql("INSERT INTO %s VALUES " +
-          "('aaa', DATE '2022-03-01', 6)," +
-          "('bbb', DATE '2022-03-01', 7)",
-          tableName);
-
-      TestHelpers.assertRows(
-          sql("SELECT * FROM %s", tableName),
-          Lists.newArrayList(Row.of("aaa", dt, 6), Row.of("bbb", dt, 7)));
+      testUpsert(tableName, row -> Row.of(row.getField(1), dt, row.getField(0)));
```
Comment (kbendick) [inline flink/v1.13/flink/src/test/java/org/apache/iceberg/flink/TestFlinkUpsert.java:192]: In my experience, when tests have failed, when of the indicators is that one or more of the resulting `Row`s will have extra fields.

So is mapping into specific Row fields possibly making it more likely that we will miss those things?

**Comment by kbendick** [comment]
Here's a link to the unit test shared code that openinx had mentioned would be good to clean up / reuse across several tests: https://github.com/apache/iceberg/pull/4364#discussion_r833870953

**Comment by yittg** [comment]
@kbendick feel free to ping me, will be glad if i can help.

And closing this PR.

**Pull request closed without merging**